transformer/events_creator.go

```diff
@@ -65,12 +65,13 @@ func (t *Transformer) CreateBeatEvents(ctx context.Context, eventData evaluator.
 			Meta:      t.eventMetadata,
 			Timestamp: timestamp,
 			Fields: mapstr.M{
-				"resource":    resource,
-				"resource_id": resMetadata.ID,   // Deprecated - kept for BC
-				"type":        resMetadata.Type, // Deprecated - kept for BC
-				"cycle_id":    eventData.CycleId,
-				"result":      finding.Result,
-				"rule":        finding.Rule,
+				resMetadata.Type: eventData.GetElasticCommonData(),
```

Comment:
@kfirpeled  - As this field has no mapping, ES index it dynamically.
I guess that we are fine with it as the purpose of this change is the ability to match data across multiple features.
Plz, correct me if I'm wrong.

Reply:
> I'm not sure I'm following the code example here, What are the values for `resMetadata.Type`? do we make sure it is only `file` & `process`?
>

At the moment, only the file & process fetchers implement the `GetElasticCommonData` and their types are `file` and `process` accordingly.
Once other fetchers will support it we can modify this logic and add a switch case or something similar.

Regarding the ECS mapping, it seems that it's not being applied by default [(discussion)](https://discuss.elastic.co/t/ecs-fields-not-mapping-correctly-automatically/224190/3), and the mapping needed to be imported explicitly [(fields.ecs.yml)](https://github.com/elastic/ecs/blob/8.3/generated/beats/fields.ecs.yml).